Appful/Token.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Token.swift
+//  Appful
+//
+//  Created by Karl Rivest Harnois on 16-04-27.
+//  Copyright © 2016 Dynamo. All rights reserved.
+//
+
+import Foundation
+
+//
+// API
+//
+
+public func fetchToken(email: String,
+                     _ password: String,
+                     _ completionHandler: JsonTaskCompletionHandler) -> NSURLSessionDataTask? {
+    return authenticationRequest(email, password).flatMap{ request in
```

Comment:
yes

it takes a closure, if the optional value being flatMapped is non-nil it passes it in the closure and rewrap it as an optional. Thus you can chain closures with flatMaps and only have to check for nil at the end.


________________________________
De : Hugo Bastien <notifications@github.com>
Envoyé : 29 avril 2016 07:52
À : hugobast/Appful
Cc : Karl Rivest Harnois; Author
Objet : Re: [hugobast/Appful] Authentication (#2)


In Appful/Token.swift<https://github.com/hugobast/Appful/pull/2#discussion_r61565514>:

> +//  Appful
> +//
> +//  Created by Karl Rivest Harnois on 16-04-27.
> +//  Copyright © 2016 Dynamo. All rights reserved.
> +//
> +
> +import Foundation
> +
> +//
> +// API
> +//
> +
> +public func fetchToken(email: String,
> +                     _ password: String,
> +                     _ completionHandler: JsonTaskCompletionHandler) -> NSURLSessionDataTask? {
> +    return authenticationRequest(email, password).flatMap{ request in


I struggled with understanding this use of .flatMap but I think I understand now. Essentially this is unwrapping the optional is that more or less accurate?

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/hugobast/Appful/pull/2/files/29ae6e7d061f2455f404dd97ea2adab2967e492d#r61565514>